docs/contracts/transient-storage.rst

```diff
@@ -0,0 +1,152 @@
+
+When transient storage is used in the context of ``DELEGATECALL`` or ``CALLCODE``,
+then the owning contract of the transient storage is the contract that issued ``DELEGATECALL``
+or ``CALLCODE`` instruction (the caller) as with persistent storage.
+When transient storage is used in the context of ``CALL`` or ``STATICCALL``,
+then the owning contract of the transient storage is the contract that is the target
+of the ``CALL``or ``STATICCALL`` instruction (the callee).
+
+.. note::
+    In the case of ``DELEGATECALL``, since references to transient storage variables
+    are currently not supported, it is not possible to pass those into library calls.
+    In libraries, access to transient storage is only possible using inline assembly.
+
+If a frame reverts, all writes to transient storage that took place between entry
+to the frame and the return are reverted, including those that took place in inner calls.
+The caller of an external call may employ a ``try ... catch`` block to prevent reverts
+bubbling up from the inner calls.
+
+*********************************************************************
+Composability of Smart Contracts and the Caveats of Transient Storage
+*********************************************************************
+
+Given the caveats mentioned in the specification of EIP-1153,
+in order to preserve the composability of your smart contract,
+utmost care is recommended for more advanced use cases of transient storage.
+
+For smart contracts, composability is a very important design to achieve a self-contained behaviour,
+such that multiple calls into individual smart contracts can be composed to more complex applications.
+So far the EVM largely guaranteed composable behaviour, since multiple calls into a smart contract
+within a complex transaction are virtually indistinguishable from multiple calls to the contract
+stretched over several transactions. However, transient storage allows a violation to this principle
+and incorrect use may lead to complex bugs that only surface when used across several calls.
+
+Let's illustrate the problem with a simple example:
+
+.. code-block:: solidity
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.27;
+
+    contract MulService {
+        uint transient multiplier;
+        function setMultiplier(uint mul) external {
+            multiplier = mul;
+        }
+
+        function multiply(uint value) external view returns (uint) {
+            return value * multiplier;
+        }
+    }
+
+If the example used memory or storage to store the multiplier, it would be fully composable.
+It would not matter whether you split the sequence into separate transactions or grouped them in some way.
+You would always get the same result. This enables use cases such as batching calls from multiple transactions
+together to reduce gas costs. Transient storage potentially breaks such use cases since composability can no longer be taken for granted.
```

Comment:
You're mentioning the call sequence from the blog post, but not showing it :) I think this paragraph will be hard to understand without it.

Reply:
I probably chopped it our when combining the texts :)